### PR TITLE
Allows running vessel in directories nested underneath a project

### LINKED ISSUE #24: vessel does not work in subdirectories (and error message is not helpful)
```
~/dfinity/motoko-merkle-tree/gen-tests $ vessel bin
Error: Failed to parse the package set file

Caused by:
    No such file or directory (os error 2)
~/dfinity/motoko-merkle-tree/gen-tests $ ls
dist-newstyle  gen-tests.cabal  main.hs
~/dfinity/motoko-merkle-tree/gen-tests $ ls ../package-set.dhall 
../package-set.dhall
```

I would have expected `vessel` to work in a subdirectory as well.

If that is intentional, the error message could be better, e.g. say

> Did not find `package-set.dhall`. `vessel` must be run in the same directory as the `package-set.dhall`.

### LINKED ISSUE #25: Search dominating vessel.dhall file
At the moment all vessel commands need to be run in the project root. I'd like for `vessel` to search for `vessel.dhall` in parent directories so you can run it from within a `test/` or `src/` directory.

`cargo` and `npm` seem to do the same thing.

We should still download packages and binaries into a `.vessel` directory next to the `vessel.dhall` file, but output paths relative to where `vessel` was invoked.

#24 shows that this is an actual need :)

### PR DESCRIPTION
**Pull request by kritzcreek**
Closes #24, Closes #25

### CODE REVIEW

**Comment by crusso** [comment]
Probably doesn't matter but is it possible to encounter cycles via symlinks and woyld you then fail to terminate? Maybe add a simple cutoff, if you are at all worried.

File: src/lib.rs [hunk 1]
```diff
@@ -18,11 +19,44 @@ use walkdir::WalkDir;
...
 impl Vessel {
+    fn find_dominating_manifest() -> Result<Option<u32>> {
+        let mut cwd = env::current_dir().context("Unable to access the current directory")?;
+        let mut nested = 0;
+        loop {
+            if cwd.join("vessel.dhall").exists() {
+                if nested != 0 {
+                    info!("Changing working directory to {}", cwd.display());
+                    env::set_current_dir(&cwd).context(format!(
+                        "Failed to change current directory to {}",
```
Comment (crusso) [inline src/lib.rs:35]: this will construct to error message even if unused.

**Review by crusso — APPROVED** [review]
LGTM

File: src/lib.rs (hunk 1 above)
Comment (kritzcreek) [reply src/lib.rs:35]: I'm not too worried about that, this is run exactly once per invocation.

**Comment by kritzcreek** [comment]
> Probably doesn't matter but is it possible to encounter cycles via symlinks and woyld you then fail to terminate? Maybe add a simple cutoff, if you are at all worried.

Since I'm only ever moving "up" the directory tree... how'd you set that scenario up? You'd need to somehow set up a symlink that points downward, while also being the parent of the thing you're pointing to...

**Pull request merged**